avif-coder/build.gradle

```diff
@@ -1,75 +1,70 @@
-plugins {
-    id 'com.android.library'
-    id 'org.jetbrains.kotlin.android'
-    id 'maven-publish'
-}
+@file:Suppress("UnstableApiUsage")
 
-task androidSourcesJar(type: Jar) {
-    archiveClassifier.set('sources')
-    from android.sourceSets.main.java.srcDirs
+plugins {
+    id("com.android.library")
+    id("org.jetbrains.kotlin.android")
 }
 
-afterEvaluate {
-    publishing {
-        publications {
-            mavenJava(MavenPublication) {
-                groupId = 'com.github.awxkee'
-                artifactId = 'avif-coder'
-                version = '1.0.17'
-                from components.release
-                artifact androidSourcesJar
-            }
-        }
-    }
-}
 
 android {
-    namespace 'com.github.awxkee.avifcoder'
-    compileSdk 33
+    namespace = "com.github.awxkee.avifcoder"
+    compileSdk = 34
 
     defaultConfig {
-        minSdk 24
-        targetSdk 33
+        minSdk = 24
+        targetSdk = 34
 
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-        consumerProguardFiles "consumer-rules.pro"
         externalNativeBuild {
             cmake {
-                cppFlags ""
-                abiFilters 'x86_64', 'armeabi-v7a', 'arm64-v8a'
-                ndkVersion '25.1.8937393'
+                cppFlags(
+                    "-Wl,--build-id=none",
+                    "-Wl,--gc-sections",
+                    "-ffunction-sections",
+                    "-fdata-sections",
+                    "-fvisibility=hidden"
+                )
+                cFlags(
+                    "-ffunction-sections",
+                    "-fdata-sections",
+                    "-fvisibility=hidden",
+                    "-Wl,--build-id=none",
+                    "-Wl,--gc-sections",
+                )
+                abiFilters += setOf("armeabi-v7a", "arm64-v8a", "x86_64")
```

Comment:
Build ndkVersion must be clearly set because not all the NDK versions have support of required configurations